### PR TITLE
Grey out button for invalid team member fields

### PR DESCRIPTION
**Pull request by rickymao**
[Notion ticket](https://www.notion.so/uwblueprintexecs/Resource-management-new-team-member-Done-button-is-not-greyed-out-089cc401605a4125910cfc313d469c32)

[Figma link](https://www.figma.com/file/IpOPyv79gBwEohvOJvPV6f/%F0%9F%9A%91-Designs-for-Development-S20-F20?node-id=1972%3A35883)

## Changes
- Added checks for valid form fields when adding a new team member
- Field checks for member name and valid email
- Added a way to get validator results from the component FormFields
- Default the role selection to Supervisor

## Screenshots
Screen Shot 2020-11-19 at 12 12 03 PM
Screen Shot 2020-11-19 at 12 12 24 PM

## Testing
- Open to the team member add flow
- Try the following combination to see if valid checks work:
> Empty name field & valid email field = INVALID
> filled name field & invalid email field (invalid email or email in use) = INVALID
> filled name field & valid email field = VALID

VALID = coloured button, INVALID = greyed out button

before opening PR
- [x] check notion ticket
- [x] run linter
- [x] go through file diff

filling out PR
- [x] descriptive title
- [x] update `notion ticket` link
- [x] update `figma link`
- [x] fill out `Changes`
  - [ ] \(optional) add inline comments in diff
- [x] fill out `Testing`
- [x] \(optional) add `Screenshots`
- [x] assign yourself to the PR

after opening PR
- [x] link PR to notion ticket
- [x] move ticket to `In PR`
- [x] make sure build passes
- [x] ping `@ps` in `#paramedics-dev`

### CODE REVIEW

File: src/components/Maps/SearchBar.tsx [hunk 1]
```diff
@@ -116,16 +116,17 @@ const SearchBar = ({
...
-                  key={suggestion.description}
-                  classes={{ root: styles.sidebarOptions }}
-                >
-                  <div {...getSuggestionItemProps(suggestion)}>
-                    <Typography>{suggestion.description}</Typography>
-                  </div>
-                </ListItem>
-              ))}
+              {!loading &&
+                suggestions.map((suggestion) => (
```
Comment (petorsong) [inline src/components/Maps/SearchBar.tsx]: was this a prettier change? let's not touch files that have nothing to do with your PR if possible

File: src/components/common/FormField.tsx [hunk 2]
```diff
@@ -47,6 +47,7 @@ const FormField: React.FC<{
   readOnly?: boolean;
   disabled?: boolean;
   numeric?: boolean;
+  whenValid?: (boolean) => void;
```
Comment (petorsong) [inline src/components/common/FormField.tsx]: if we're going to put another prop on `FormField`, i'd like to follow some naming convention - would `onValidate` make sense here? whatever you choose will need corresponding changes in `UserFormPage` as well

Comment (petorsong) [reply src/components/common/FormField.tsx]: also how does this differ from `isValidated` and `validators`? is it necessary to have all 3 props?

cc @aaronabraham311 - i think he was the one who added these originally

**Review by petorsong — CHANGES_REQUESTED** [review]
i take rapid PR review requests at competitive prices - DM me for deets 💰

File: src/components/Maps/SearchBar.tsx (hunk 1 above)
Comment (JasonYG) [reply src/components/Maps/SearchBar.tsx]: +1 to Peter's comment

Comment (JasonYG) [reply src/components/Maps/SearchBar.tsx]: maybe this was changed by `npm run lint`?

File: src/components/common/FormField.tsx (hunk 2 above)
Comment (JasonYG) [reply src/components/common/FormField.tsx]: +1 to changing to `onValidate`. Common practice is to name event handlers with `onSomething`, while the function you pass in would be `handleSomething`.

**Review by petorsong — CHANGES_REQUESTED** [review]
code looks fine but attempting to edit a user takes you to the edit flow but with no role selected - this may be a bug introduced by unrelated changes but i think it's worth looking into as part of this PR (sorry for scope creep 😬 )

**Comment by rickymao** [comment]
> code looks fine but attempting to edit a user takes you to the edit flow but with no role selected - this may be a bug introduced by unrelated changes but i think it's worth looking into as part of this PR (sorry for scope creep 😬 )

Should be fixed now 👍

**Review by petorsong — DISMISSED** [review]

**Pull request merged**